gedcom_view.py

```diff
@@ -24,13 +24,14 @@ def green(s):
 
 if len(sys.argv) == 1:
     gedcom = sys.stdin
+    buffer = gedcom.read().replace('\r\n', '\n').replace('\r', '\n')
 else:
-    gedcom = open(sys.argv[1], "r")
+    with open(sys.argv[1], 'r', buffering=1) as gedcom:
+        buffer = gedcom.readlines()
```

Comment:
This is still allocating a lot of memory, since we are storing the whole file, albeit in a better way maybe.
I think we could do without `buffer` here, and directly use `for line in gedcom` below. 
Some long term plans is to use `mmap` to read the file, parse it to find the start of top-level (level 0) records. We then need to parse them in the right order (sources first, if I remember right), so that we can then parse the others, without having to keep the whole contents of the file in memory. Currently, we need to store `buffer` in memory, and then the result of splitting the lines, so basically we have the whole file twice in memory...

Reply:
Absolutely. This fix is merely intended to allow much larger files to actually be read. It doesn't address that they _are_ much larger files and eat a lot of memory. As far as those longer term plans, perhaps `diskcache` might be worth considering.

Also, I've removed the `buffer` usage, and now read directly from stdin or use `readlines()` to bring in the file, with `gedcom`.

Reply:
As a possible tool, if you aren't familiar with it:

http://www.grantjenks.com/docs/diskcache/tutorial.html